base/sort.jl

```diff
@@ -407,6 +408,48 @@ julia> searchsortedlast([1, 2, 4, 5, 5, 7], 0) # no match, insert at start
 ```
 """ searchsortedlast
 
+function insorted(x, coll::AbstractVector, lo::T, hi::T, o::Ordering)::Bool where T<:Integer
+    !isempty(searchsorted(coll, x, lo, hi, o))
+end
+
+insorted(x, r::AbstractRange) = in(x, r)
+
+for s in [:insorted]
+    @eval begin
+        $s(x, v::AbstractVector, o::Ordering) = (inds = axes(v, 1); $s(x,v,first(inds),last(inds),o))
+        $s(x, v::AbstractVector;
```

Comment:
This is the method we want. But why restrict on `AbstractVector`? Also why put it into an `@eval` loop?

Reply:
As `searchsorted` is only working on `AbstractVector` and `AbstractRange` I though it would be better to restrict it here.
Not doing so won't lead to something like the following?
```julia
insorted(5, 6)
ERROR: No method matching searchsorted(Int, Int)
```

Reply:
If users overload `searchsorted` for their own containers, they will get `insorted` for free. If we restrict to `AbstractVector` the user container must be `AbstractVector` to get the free method. OTOH code or errors may be clearer with the restriction. I don't have a strong opinion, do as you see fit.

Reply:
Thanks for the precision, now I get why it could be interesting to remove this kind of restriction.